app/controllers/application_controller.rb

```diff
@@ -958,4 +959,25 @@ def activated_themes_json
     ids = Theme.transform_ids(id)
     Theme.where(id: ids).pluck(:id, :name).to_h.to_json
   end
+
+  def rate_limit_crawlers
+    return if current_user.present?
+    user_agent = request.user_agent&.downcase
+    return if user_agent.blank?
+
+    SiteSetting.slow_down_crawler_user_agents.downcase.split("|").each do |crawler|
```

Comment:
no need to downcase and split a blank, I would just early return prior to user_agent check even if slow_down_crawler_user_agents.blank?